Python/bytecodes.c

```diff
@@ -3945,16 +3957,21 @@ dummy_func(
 
         op(_SET_IP, (--)) {
             TIER_TWO_ONLY
-            frame->prev_instr = ip_offset + oparg;
+            frame->instr_ptr = ip_offset + oparg;
         }
 
         op(_SAVE_CURRENT_IP, (--)) {
             TIER_ONE_ONLY
-            frame->prev_instr = next_instr - 1;
+            if (frame->next_instr_offset == 0) {
+                frame->next_instr_offset = next_instr - frame->instr_ptr;
+            }
+            else {
+                assert(next_instr == frame->instr_ptr);
+            }
```

Comment:
I'm not sure why I needed to do it this way (what the else case is doing).

Reply:
Checking again, this is no longer needed to it was there to work around something I've already fixed.